plugin/bind/setup.go

```diff
@@ -84,15 +84,23 @@ func listIP(args []string, ifaces []net.Interface) ([]string, error) {
 				}
 				for _, addr := range addrs {
 					if ipnet, ok := addr.(*net.IPNet); ok {
-						if ipnet.IP.To4() != nil || (!ipnet.IP.IsLinkLocalMulticast() && !ipnet.IP.IsLinkLocalUnicast()) {
-							all = append(all, ipnet.IP.String())
+						ipa, err := net.ResolveIPAddr("ip", ipnet.IP.String())
+						if err == nil {
+							if len(ipnet.IP) == net.IPv6len &&
+								(ipnet.IP.IsLinkLocalMulticast() || ipnet.IP.IsLinkLocalUnicast()) {
+								if ipa.Zone == "" {
+									ipa.Zone = iface.Name
+								}
+							}
+							all = append(all, ipa.String())
 						}
 					}
 				}
 			}
 		}
 		if !isIface {
-			if net.ParseIP(a) == nil {
+			_, err := net.ResolveIPAddr("ip", a)
```

Comment:
Thanks for catching this. 

Hadn't thought of this.

 My suggestion: I revert that second part, because if an interface is given this can be guaranteed. 
I am using this function only because it maintains the "zone" (aka scope).  And then the problem can be split off into another PR, as this is actually independent of the first part.

However, I wonder what the adverse effects could be (Attack scenario: provide a hostname that mimics an interface but provide a different IP), as the bind should fail and this is only a function for error handling. The only alternative I think would be to reimplement the scope parsing completely?!

ideas:
1. I could limit the impact if I only resort to ResolveIPAddr if ParseIP fails.
2. I could stringify the address again and do an equality check to make sure no resolving happens.
3. Find a better go function? Any ideas?

Reply:
Or  declare it a feature that you now can also bind "localhost" ;)